app/src/main/java/org/p2p/wallet/settings/ui/newreset/pin/NewForgotPinBottomSheet.kt

```diff
@@ -0,0 +1,48 @@
+package org.p2p.wallet.settings.ui.newreset.pin
+
+import androidx.core.os.bundleOf
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.setFragmentResult
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import org.p2p.wallet.R
+import org.p2p.wallet.common.ui.NonDraggableBottomSheetDialogFragment
+import org.p2p.wallet.databinding.DialogNewForgotPasswordBinding
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+
+private const val EXTRA_REQUEST_KEY = "EXTRA_REQUEST_KEY"
+private const val EXTRA_RESULT_KEY = "EXTRA_RESULT_KEY"
+
+class NewForgotPinBottomSheet : NonDraggableBottomSheetDialogFragment() {
+
+    companion object {
+        fun show(fm: FragmentManager, requestKey: String, resultKey: String) =
+            NewForgotPinBottomSheet().withArgs(
+                EXTRA_REQUEST_KEY to requestKey,
+                EXTRA_RESULT_KEY to resultKey
+            ).show(fm, NewForgotPinBottomSheet::javaClass.name)
+    }
+
+    private val binding: DialogNewForgotPasswordBinding by viewBinding()
+    private val requestKey: String by args(EXTRA_REQUEST_KEY)
+    private val resultKey: String by args(EXTRA_RESULT_KEY)
+
+    override fun getTheme(): Int = R.style.WalletTheme_BottomSheet_Rounded
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
+        inflater.inflate(R.layout.dialog_new_forgot_password, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+            buttonLogout.setOnClickListener {
+                setFragmentResult(requestKey, bundleOf(resultKey to true))
```

Comment:
im not sure what is `true` here
we send true when button is clicked? or user confirmed logout? or he is confirming that he forgot password / pin?

maybe we should rename resultKey to provide some intent what value he holds inside? 
smth like resultKeyLogoutConfirmed
this bottom sheet is not universal, we can add some context to request and result key naming

Reply:
We are not using the value, I just need to send some bundle to check if user clicked on logout button

Reply:
how about a Unit?

Reply:
> how about a Unit?

first, I passed Unit, but it crashed. Looks like we cannot pass Unit in bundle